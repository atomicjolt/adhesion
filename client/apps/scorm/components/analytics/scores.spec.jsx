import React from 'react';
import TestUtils from 'react-addons-test-utils';
import Stub from '../../../../specs_support/stub';
import Scores from './scores';

describe('Scorm Analytics Scores', () => {

  let result;
  const props = {
    scores: [
      {
        name: 'test application',
        value: '12',
      },
    ],
  };

  beforeEach(() => {
    result = TestUtils.renderIntoDocument(
      <Stub>
        <Scores {...props} />
      </Stub>
    );
  });

  it('renders the scores with the correct values', () => {
    const div =  TestUtils.findRenderedDOMComponentWithClass(result, 'c-aa-label');
    expect(div.textContent).toContain(props.scores[0].name);
    expect(div.textContent).toContain(props.scores[0].value);
  });
});
