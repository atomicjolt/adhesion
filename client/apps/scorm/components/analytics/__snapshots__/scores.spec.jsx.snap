// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scorm Analytics Scores matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AverageScore
    scores={
      Array [
        Object {
          "name": "test application",
          "value": "12",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ScoreLabel
          name="test application"
          value="12"
        />,
      ],
      "style": Object {
        "position": "absolute",
        "right": "5rem",
        "top": "2rem",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "scores_test application",
        "nodeType": "function",
        "props": Object {
          "name": "test application",
          "value": "12",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ScoreLabel
            name="test application"
            value="12"
          />,
        ],
        "style": Object {
          "position": "absolute",
          "right": "5rem",
          "top": "2rem",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "scores_test application",
          "nodeType": "function",
          "props": Object {
            "name": "test application",
            "value": "12",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
